tendermint/spn
# feat(`claim`): add hooks for delegation and vote mission claim

Implement hooks for `staking` and `gov` modules that allow to complete claim mission on the events:
- Delegating funds for `staking`
- Voting on proposal for `gov`

Set hooks in `app.go` as a sample implementation.

Security mechanism remains to be designed: locking a specific mission ID for a specific module

Add event on mission completion

### To test:
Serve:
```
ignite chain serve -r
```

Claim for delegation mission:
```
spnd tx staking delegate spnvaloper1ezptsm3npn54qx9vvpah4nymre59ykr9mx22g4 1500uspn --from bob -y
```

Claim for voting proposal:
```
spnd tx gov submit-proposal cancel-software-upgrade --title foo --description bar --from alice -y
spnd tx gov deposit 1 1uspn --from alice -y
spnd tx gov vote 1 yes --from bob -y
```

Check balance for `drop` token:
```
spnd q bank balances spn1aqn8ynvr3jmq67879qulzrwhchq5dtrvtx0nhe
```

Check claim records:
```
spnd q claim list-claim-record
```
